tabular/src/autogluon/tabular/models/tab_transformer/pretexts.py

```diff
@@ -0,0 +1,162 @@
+from copy import deepcopy
+import numpy as np
+
+from ...constants import REGRESSION
+from ...try_import import try_import_torch
+
+"""
+possible TODO: although there is a supervised pretext option below, i.e. pretrain using
+labeled data presumably from a different but related task, curently this cannot be handled 
+by the AG API. One simple way to handle it would be to have an optional "pretext label" 
+argument to task.fit that tells the pretraining module to use the supervised pretext task
+with that particular column as the label.
+"""
+class PretextClass:
+    try_import_torch()
```

Comment:
so long as this file is only imported after checking torch exists, you don't need these try import statements and can just import from the top of file.

Reply:
Compressed all the lazy imports down to one in TabTransformerModel class.